provider/pkg/gen/discovery.go

```diff
@@ -84,11 +119,15 @@ func findResourcesImpl(docName, parentName string, rest map[string]discovery.Res
 				typeName := fmt.Sprintf("%s%sIamPolicy", parentName, name)
 				if getIamPolicy, has := res.Methods["getIamPolicy"]; has {
 					dd := discoveryDocumentResource{
-						createMethod: &restMethod,
-						getMethod:    &getIamPolicy,
-						updateMethod: &restMethod,
+						discoveryCRUDMethods: discoveryCRUDMethods{
+							createMethod: &restMethod,
+							getMethod:    &getIamPolicy,
+							updateMethod: &restMethod,
+							listMethod:   listMethod,
```

Comment:
Should this be a reference? `&listMethod`

Reply:
listMethod is actually a pointer anyway.